spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -359,8 +359,8 @@ private boolean canDeleteUsingMetadata(Expression deleteExpr) {
   }
 
   @Override
-  public void deleteWhere(Filter[] filters) {
-    Expression deleteExpr = SparkFilters.convert(filters);
+  public void deleteWhere(Predicate[] predicates) {
+    Expression deleteExpr = SparkV2Filters.convert(predicates);
```

Comment:
So this is the one case where we use convert on a list of predicates instead one at a time, with canDeleteWhere using a for loop on the individual expressions we will can different results.

I think given the use case of convert + list of predicates we should change that to not throw as originally suggest by @rdblue .

Reply:
Please see the above answers. I think here is deliberate.

Reply:
Gotcha, I think I missed the `this method will be invoked only if {@link #canDeleteWhere(Predicate[])} returns true.` (I was looking at the Iceberg trait this extends instead of the upstream source).